pybamm/models/base_model.py

```diff
@@ -585,6 +683,17 @@ def build_coupled_variables(self):
                         )
                     )
                     try:
+                        model_var_copy = self.variables.copy()
+                        submodel_coupled_result = submodel.get_coupled_variables(
```

Comment:
not sure I like the name `submodel_coupled_result` maybe just `updated_variables` since it contains _all_ the variables up to this point?

Reply:
Sure! `updated_variables` sounds better